robot_sf/render/sim_view.py

```diff
@@ -91,7 +91,7 @@ class SimulationView:
     focus_on_robot: bool = False
     focus_on_ego_ped: bool = False
     display_help: bool = False
-    display_robot_info: bool = False
+    display_robot_info: int = 0
```

Comment:
Do you think we could replace this with an Enum, or do you think this would add complexity, where it is not needed?

To improve the `display_robot_info` attribute, you can replace the integer-based approach with an enumeration. This makes the code more readable and maintainable by providing explicit names for the different cases.

Here's how you can do it:

1. Define an enumeration for the different display modes.
2. Update the `SimulationView` class to use this enumeration.

First, define the enumeration in a suitable location, such as at the top of the 

sim_view.py

 file:

```python
from enum import Enum

class DisplayMode(Enum):
    NONE = 0
    ROBOT_INFO = 1
    PEDESTRIAN_INFO = 2
```

Next, update the `SimulationView` class to use this enumeration:

```python
@dataclass
class SimulationView:
    ...
    display_robot_info: DisplayMode = DisplayMode.NONE
    ...
    def _handle_keydown(self, e):
        ...
        key_action_map = {
            ...
            pygame.K_q: lambda: setattr(self, 'display_robot_info',
                                        DisplayMode((self.display_robot_info.value + 1) % len(DisplayMode))),
            ...
        }
        ...
    def _add_text(self, timestep: int, state: VisualizableSimState):
        lines = []
        if self.display_robot_info == DisplayMode.ROBOT_INFO and state.robot_action:
            lines += [
                f'RobotPose: {state.robot_pose}',
                f'RobotAction: {state.robot_action.action if state.robot_action else None}',
                f'RobotGoal: {state.robot_action.goal if state.robot_action else None}',]
        elif self.display_robot_info == DisplayMode.PEDESTRIAN_INFO:
            if state.ego_ped_pose and state.ego_ped_action:
                distance_to_robot = euclid_dist(state.ego_ped_pose[0], state.robot_pose[0])
                lines += [
                    f'PedestrianPose: {state.ego_ped_pose}',
                    f'PedestrianAction: {state.ego_ped_action.action}',
                    f'PedestrianGoal: {state.ego_ped_action.goal}',
                    f'DistanceRobot: {distance_to_robot:.2f}',]
            else:
                self.display_robot_info = DisplayMode.NONE
        ...
```

This change makes the code more explicit and easier to understand. The `DisplayMode` enumeration clearly defines the possible display modes, and the code that uses `display_robot_info`

Reply:
Hard to tell, in my opinion it adds complexity. But i can see how this change might help more unexperienced users.

It also might be useful, if more states to display are introduced and the _add_text function gets split up.

If this change is introduced i would name the enum to TextMode, so DisplayMode can be used to activate/deactivate the grid or some debugging objects with keyevents (just like switching through the Text or with specific keys like 1,2,3,...)